hedera-node/src/main/java/com/hedera/services/context/AwareTransactionContext.java

```diff
@@ -310,14 +313,24 @@ public void payerSigIsKnownActive() {
 
 	@Override
 	public void trigger(TxnAccessor scopedAccessor) {
-		if (this.accessor().isTriggeredTxn()) {
+		if (accessor().isTriggeredTxn()) {
 			throw new IllegalStateException("Unable to trigger txns in triggered txns");
 		}
-		this.triggeredTxn = scopedAccessor;
+		triggeredTxn = scopedAccessor;
 	}
 
 	@Override
 	public TxnAccessor triggeredTxn() {
 		return triggeredTxn;
 	}
+
+	@Override
+	public void setExpiringEntity(ExpiringEntity entity) {
```

Comment:
I'd probably delay adding these methods---that is, `setExpiringEntity()` and `expiringEntity()`---until we have more clarity how how to generalize expiration for all entity types. 

Notice, for example, that a single `ContractCall` transaction can create many accounts.